src/windows.rs

```diff
@@ -1,18 +1,30 @@
 use super::{Height, Width};
+use std::os::windows::io::RawHandle;
 
-/// Returns the size of the terminal, if available.
+/// Returns the size of the terminal defaulting to STDOUT, if available.
 ///
 /// Note that this returns the size of the actual command window, and
 /// not the overall size of the command window buffer
 pub fn terminal_size() -> Option<(Width, Height)> {
-    use winapi::um::handleapi::INVALID_HANDLE_VALUE;
     use winapi::um::processenv::GetStdHandle;
     use winapi::um::winbase::STD_OUTPUT_HANDLE;
+
+    let handle = unsafe { GetStdHandle(STD_OUTPUT_HANDLE) as RawHandle };
+
+    terminal_size_using_handle(handle)
+}
+
+/// Returns the size of the terminal using the given handle, if available.
+///
+/// If the given handle is not a tty, returns `None`
+pub fn terminal_size_using_handle(handle: RawHandle) -> Option<(Width, Height)> {
+    use winapi::um::handleapi::INVALID_HANDLE_VALUE;
     use winapi::um::wincon::{
         GetConsoleScreenBufferInfo, CONSOLE_SCREEN_BUFFER_INFO, COORD, SMALL_RECT,
     };
 
-    let hand = unsafe { GetStdHandle(STD_OUTPUT_HANDLE) };
+    // convert between winapi::um::winnt::HANDLE and std::os::windows::raw::HANDLE
+    let hand = unsafe { std::mem::transmute(handle) };
```

Comment:
Transmute scares me.  Can't we just cast this?

```suggestion
    let hand = handle as winapi::um::winnt::HANDLE;
```

Reply:
I am not sure. It has existed in [console](https://github.com/mitsuhiko/console/blob/2d9269c487ef56ddc184826d32b22ec1ee564a8e/src/windows_term.rs#L52) for almost 3 years since [this](https://github.com/mitsuhiko/console/commit/db5a5dde9b8d24752562b7b14ef37c1c65a21293) commit adding the native windows implementation.

Reply:
The handle is owned by this function, so, I don't think transmute will give us any issues. I am just not sure if using `as` still does the job or not.

Reply:
I dug a little bit more into this and yes, we can use `as`